mesonbuild/mtest.py

```diff
@@ -1038,6 +1039,8 @@ class TestRunExitCode(TestRun):
     def complete(self) -> None:
         if self.res != TestResult.RUNNING:
             pass
+        elif self.returncode == self.success_returncode:
+            self.res = TestResult.OK
```

Comment:
Given below we handle the `else` case as such:

```python
else:
    self.res = TestResult.FAIL if bool(self.returncode) else TestResult.OK
```

Seems like a TestResult.OK is emitted for either a returncode of 0, or a returncode of an overridden success_returncode. Is this intentional?

Reply:
That's an oversight on my part, corrected it and I added a test for this